### PR TITLE
Fix doc build on CircleCI and add ccache

### LINKED ISSUE #6645: "ubuntu-2204" breaks doc build on CircleCI
### Description:

If you have a look at https://app.circleci.com/pipelines/github/scikit-image/scikit-image, it seems like jobs using the new `ubuntu-2204` introduced in b8caa25226cb4b53724ed5234363009f115a28f9 seem broken. The failed jobs display this notification

> Job was rejected because resource class medium, image ubuntu-2204 is not a valid resource class

The ones using the old `cimg/python:3.8` seem fine which is why some PRs work and other don't, e.g. [job 4744](https://app.circleci.com/pipelines/github/scikit-image/scikit-image/4658/workflows/88840212-f302-436d-bb47-905cbe302535/jobs/4744?invite=true#step-0-13).

Pinging @grlee77 in the hope that you immediately know what's wrong. :pray: 

### Way to reproduce:

_No response_

### Traceback or output:

_No response_

### Version information:

_No response_

### PR DESCRIPTION
**Pull request by lagru**
## Description

Try to debug and fix #6645.

Link for convenience: https://app.circleci.com/pipelines/github/scikit-image/scikit-image

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Comment by lagru** [comment]
Ha, it worked! However, it currently has to build scikit-image for each run. I'll try to copy the ccache approach from SciPy. 

@grlee77 Do you remember why you switched from `cimg/python:3.8` to `ubuntu-2204` in b8caa25226cb4b53724ed5234363009f115a28f9. Was it because of the venv / meson / docker problems described in https://github.com/scipy/scipy/issues/16312?

**Comment by lagru** [comment]
Hmm, [it didn't find anything at `~/.ccache`](https://app.circleci.com/pipelines/github/scikit-image/scikit-image/4664/workflows/aa905d4b-4bd4-4443-9ad5-8e8c102a577a/jobs/4750). I guess ccache is not correctly configured yet. But I'll wait with further tests until I get some more context...

**Comment by lagru** [comment]
Ah, the switch to `ubuntu-2204` happened in 263a7a57aa359717ab52d95e2c000084b6fbd76d. So I should rather ask you @stefanv:

> Do you remember why you switched from `cimg/python:3.8` to `ubuntu-2204` in 263a7a57aa359717ab52d95e2c000084b6fbd76d. Was it because of the venv / meson / docker problems described in https://github.com/scipy/scipy/issues/16312?

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,23 +18,20 @@ jobs:
...
-            python -m venv skimage_venv
-            source skimage_venv/bin/activate
             python -m pip install --upgrade pip
             python -m pip install -r requirements/build.txt
-            python -m pip install -vv --no-build-isolation .
             python -m pip install -r requirements/default.txt
+            python -m pip install -r requirements/optional.txt
             python -m pip install -r requirements/docs.txt
-            export
+            python -m pip install -vv --no-build-isolation .
```
Comment (hmaarrfk) [inline .circleci/config.yml:35]: The order was important to ensure that one could build without the optional dependencies

**Comment by stefanv** [comment]
> Ah, the switch to `ubuntu-2204` happened in [263a7a5](https://github.com/scikit-image/scikit-image/commit/263a7a57aa359717ab52d95e2c000084b6fbd76d). So I should rather ask you @stefanv:
> 
> > Do you remember why you switched from `cimg/python:3.8` to `ubuntu-2204` in [263a7a5](https://github.com/scikit-image/scikit-image/commit/263a7a57aa359717ab52d95e2c000084b6fbd76d). Was it because of the venv / meson / docker problems described in [scipy/scipy#16312](https://github.com/scipy/scipy/issues/16312)?

I think the docs builder ran out of RAM. Building on a Linux VM instead of Docker uses fewer resources (that's my best guess).

**Comment by mkcor** [comment]
Hi @lagru, what's the status on this?

File: .circleci/config.yml (hunk 1 above)
Comment (lagru) [reply .circleci/config.yml:35]: Good to know! Though I would have expected that kind of test elsewhere and not inside the doc build job. I'll revert it.

**Comment by lagru** [comment]
Honestly, I was out of easy ideas to try. Stéfan's comment indicates that we should go the Linux VM approach. However, I did not get that one to work due to [missing dependencies](https://app.circleci.com/pipelines/github/scikit-image/scikit-image/4661/workflows/eb5aaf01-b910-4842-a4cf-94987e00f3ef/jobs/4747). Not sure at all why venv isn't found or how to debug.

**Comment by stefanv** [comment]
Try apt-get install python3-venv maybe? Venv is not default on Debian systems for some reason.

**Review by jarrodmillman — APPROVED** [review]

**Comment by stefanv** [comment]
Thanks @lagru!

**Pull request merged**